common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -324,6 +325,11 @@ protected enum SpaceTrimming {
 
       private static final int INDETERMINATE_COLLATION_ID = -1;
 
+      /**
+       * UTF8 Binary collation version.
+       */
+      private static final String UTF8_BINARY_VERSION = "1.0";
```

Comment:
so why `1.0`?

Reply:
on another note, please see the `Lower`/`Upper` implementation in `CollationSuite`

we actually do in fact use ICU for UTF8_BINARY implementation there now as well (`execBinaryICU`)
so I suggest putting the proper ICU version here